### PR TITLE
[intel-npu] removing cpu_pinning string from CID command string

### PR DESCRIPTION
**Pull request by csoka**
### Details:
 - removing ENABLE_CPU_PINNING from command string passed to CiD, for older compiler versions to enable backwards compatibility

### Tickets:
 - EISW-124233

### CODE REVIEW

**Review by sbutnari — APPROVED** [review]

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 1]
```diff
@@ -453,13 +453,19 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
         content = std::regex_replace(content, std::regex(ov::intel_npu::tiles.name()), "NPU_DPU_GROUPS");
     }
 
-    // Batch mode property is not supported in versions < 5.5 - need to remove it
+    // Batch mode and cpu pinning property is not supported in versions < 5.5 - need to remove it
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:456]:
```suggestion
    // Batch mode and cpu pinning properties are not supported in versions < 5.5 - need to remove them
```

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 2]
```diff
@@ -453,13 +453,19 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
...
     if ((compilerVersion.major < 5) || (compilerVersion.major == 5 && compilerVersion.minor < 5)) {
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:457]: Is 5.5 the first version supporting this property?

Comment (csoka) [reply src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:457]: yes, verified. Actually there is a huge overlap, due to the fact that minor version wasn't bumped up when this prop was added. Prop was added last november, minor was bumped to 5.5 last month

**Review by ArtemySkrebkov — APPROVED** [review]

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp (hunk 2 above)
Comment (razvanapetroaie) [reply src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp]: Talked more in private, turns out version 5.2.0 is the right threshold.

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 3]
```diff
@@ -420,6 +410,14 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
...
                                      getTargetRegex(ov::hint::Priority::HIGH),
                                      getStringReplacement(ov::intel_npu::LegacyPriority::HIGH));
+
+        // Removing cpu_pinning from the command string
+        std::ostringstream pinningstr;
+        pinningstr << ov::hint::enable_cpu_pinning.name() << KEY_VALUE_SEPARATOR << VALUE_DELIMITER << "\\S+"
+                   << VALUE_DELIMITER;
+        _logger.warning(
+            "ENABLE_CPU_PINNING property is not suppored by this compiler version. Removing from parameters");
+        content = std::regex_replace(content, std::regex(pinningstr.str()), "");
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:420]: For anyone watching this, replacing the prefixes is best performed as the last step.

Doing this as the first step, as was the case in the previous implementation, leads to property names mismatches in the regex calls used later in the same function.

**Review by razvanapetroaie — APPROVED** [review]

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 4]
```diff
@@ -472,6 +471,19 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
...
+    if (compilerVersion.major < 5) {
+        std::regex reg("NPU_");
+        content = std::regex_replace(content, reg, "VPU_");
+        // From 4.0.0, driver compiler start to use VPU_ prefix, the old version uses VPUX_ prefix
+        if (compilerVersion.major < 4) {
+            // Replace VPU_ with VPUX_ for old driver compiler
+            std::regex reg("VPU_");
+            content = std::regex_replace(content, reg, "VPUX_");
+        }
+    }
```
Comment (razvanapetroaie) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:485]: For anyone watching this, replacing the prefixes is best performed as the last step.

Doing this as the first step, as was the case in the previous implementation, leads to property names mismatches in the regex calls used later in the same function.

**Pull request merged**